pkg/controller/linstorcsidriver/linstorcsidriver_controller_test.go

```diff
@@ -0,0 +1,323 @@
+package linstorcsidriver
+
+import (
+	"context"
+	"github.com/piraeusdatastore/piraeus-operator/pkg/apis"
+	piraeusv1alpha1 "github.com/piraeusdatastore/piraeus-operator/pkg/apis/piraeus/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	schedv1 "k8s.io/api/scheduling/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"testing"
+)
+
+var (
+	OperatorName         = "foo"
+	OperatorNamespace    = "bar"
+	DefaultNodeDaemonSet = appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo-csi-node-daemonset",
+			Namespace: "bar",
+		},
+	}
+	DefaultControllerDeployment = appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo-csi-controller-deployment",
+			Namespace: "bar",
+		},
+	}
+	DefaultNodeServiceAccount = corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo-csi-node-sa",
+			Namespace: "bar",
+		},
+	}
+	DefaulControllerServiceAccount = corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo-csi-controller-sa",
+			Namespace: "bar",
+		},
+	}
+	DefaultPriorityClass = schedv1.PriorityClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo-csi-priority-class",
+			Namespace: "bar",
+		},
+	}
+)
+
+func TestReconcileLinstorCSIDriver_Reconcile(t *testing.T) {
```

Comment:
This is really cool. We should add tests for the other controllers and whenever we fix bugs.